testing/integration/beat_receivers_test.go

```diff
@@ -201,185 +238,225 @@ func TestClassicAndReceiverAgentMonitoring(t *testing.T) {
+					mustClauses := []map[string]any{
+						{"match": map[string]any{"data_stream.type": tc.dsType}},
+						{"match": map[string]any{"data_stream.dataset": tc.dsDataset}},
+						{"match": map[string]any{"data_stream.namespace": receiverNamespace}},
+					}
+					mustClauses = append(mustClauses, tc.query...)
+
+					rawQuery := map[string]any{
+						"query": map[string]any{
+							"bool": map[string]any{
+								"must":   mustClauses,
+								"filter": map[string]any{"range": map[string]any{"@timestamp": map[string]any{"gte": timestampBeatReceiver}}},
+							},
+						},
+						"sort": []map[string]any{
+							{"@timestamp": map[string]any{"order": "asc"}},
+						},
+					}
 
-	// 7. Compare both documents are equivalent
-	for _, tc := range tests[:3] {
-		key := tc.dsType + "-" + tc.dsDataset + "-" + tc.dsNamespace
-		agent := agentDocs[key].Hits.Hits[0].Source
-		otel := otelDocs[key].Hits.Hits[0].Source
-		ignoredFields := []string{
-			// Expected to change between agentDocs and OtelDocs
-			"@timestamp",
-			"agent.ephemeral_id",
-			// agent.id is different because it's the id of the underlying beat
-			"agent.id",
-			// agent.version is different because we force version 9.0.0 in CI
-			"agent.version",
-			"elastic_agent.id",
-			"log.file.inode",
-			"log.file.fingerprint",
-			"log.file.path",
-			"log.offset",
-			"event.ingested",
+					docs, err := estools.PerformQueryForRawQuery(findCtx, rawQuery, tc.dsType+"-*", info.ESClient)
+					require.NoError(t, err)
+					if docs.Hits.Total.Value != 0 {
+						key := tc.dsType + "-" + tc.dsDataset + "-" + receiverNamespace
+						otelDocs[key] = docs
+					}
+					return docs.Hits.Total.Value > 0
+				},
+				5*time.Minute, 5*time.Second,
+				"receiver agent monitoring no documents found for timestamp: %s, type: %s, dataset: %s, namespace: %s, query: %v", timestampBeatReceiver, tc.dsType, tc.dsDataset, receiverNamespace, tc.query)
 		}
-		switch tc.onlyCompareKeys {
-		case true:
-			AssertMapstrKeysEqual(t, agent, otel, append(ignoredFields, tc.ignoreFields...), fmt.Sprintf("expected document keys to be equal for dataset: %s", key))
-		case false:
-			AssertMapsEqual(t, agent, otel, ignoredFields, fmt.Sprintf("expected document to be equal for dataset: %s", key))
+
+		// Uninstall beat receiver monitoring
+		combinedOutput, err = receiverFixture.Uninstall(ctx, &atesting.UninstallOpts{Force: true})
+		require.NoErrorf(t, err, "error uninstalling beat receiver agent monitoring, err: %s, combined output: %s", err, string(combinedOutput))
+
+		// Compare process vs beat receiver events
+		for _, tc := range tests {
+			agent := agentDocs[tc.dsType+"-"+tc.dsDataset+"-"+processNamespace].Hits.Hits[0].Source
+			otel := otelDocs[tc.dsType+"-"+tc.dsDataset+"-"+receiverNamespace].Hits.Hits[0].Source
+			ignoredFields := []string{
+				// Expected to change between agentDocs and OtelDocs
+				"@timestamp",
+				"agent.ephemeral_id",
+				"agent.id",
+				"agent.version",
```

Comment:
If we are using the translation from the otel runtime then the versions should match, you could remove this field from the ignored list.

Reply:
FYI: This field can differ. The `agent.version` will always report the version of the Beats receiver. As of today, this will likely show 9.1.0. However, if you try to backport this to 9.0, the main Agent document will report 9.0, while the Beats receiver will still report 9.1.0.